src/ape_ethereum/ecosystem.py

```diff
@@ -216,19 +217,21 @@ def str_to_slot(text):
 
             return ProxyInfo(type=type, target=target)
 
-        # gnosis safe stores implementation in slot 0, read `masterCopy()` to be sure
+        # gnosis safe stores implementation in slot 0, read `NAME()` to be sure
         abi = MethodABI(
             type="function",
-            name="masterCopy",
+            name="NAME",
             stateMutability="view",
-            outputs=[ABIType(type="address")],
+            outputs=[ABIType(type="string")],
         )
         try:
-            master_copy = ContractCall(abi, address)()
-            storage = self.provider.get_storage_at(address, 0)
-            slot_0 = self.conversion_manager.convert(storage[-20:].hex(), AddressType)
-            if master_copy == slot_0:
-                return ProxyInfo(type=ProxyType.GnosisSafe, target=master_copy)
+            name = ContractCall(abi, address)()
+            raw_target = self.provider.get_storage_at(address, 0)[-20:].hex()
+            target = self.conversion_manager.convert(raw_target, AddressType)
+            # NOTE: `target` is set in initialized proxies
+            if name in ("Gnosis Safe", "Default Callback Handler") and target != ZERO_ADDRESS:
```

Comment:
could compare raw_target to zero address and only do the AddressType conversion within this if stmt, as a very minor optimization

Reply:
I think that you'd need to do the conversion to get the comparison to work properly